package/sbp_nmea2000_bridge/src/sbp_nmea2000_bridge.cpp

```diff
@@ -0,0 +1,917 @@
+                          tN2kHeadingReference::N2khr_true,
+                          cog_rad, sog_mps);
+      } else {
+        SetN2kCOGSOGRapid(N2kMsg, /*sequence ID=*/0xFF,
+                          tN2kHeadingReference::N2khr_true,
+                          N2kDoubleNA, N2kDoubleNA);
+      }
+      NMEA2000.SendMsg(N2kMsg);
+
+      d << "\tDone.\n";
+    }
+
+    // PGN 129539
+    void callback_sbp_dops(u16 sender_id, u8 len, u8 msg[], void *context){
+      UNUSED(sender_id);
+      UNUSED(len);
+      UNUSED(context);
+      d << __FUNCTION__ << "\n";
+
+      auto sbp_dops = reinterpret_cast<msg_dops_t*>(msg);
+
+      bool is_valid = is_flags_valid(sbp_dops->flags, /*bits=*/3);
+
+      tN2kMsg N2kMsg;
+      if (is_valid) {
+        SetN2kGNSSDOPData(N2kMsg, tow_sequence_id_cache[sbp_dops->tow],
+                /*DesiredMode=*/tN2kGNSSDOPmode::N2kGNSSdm_Auto,
+                /*ActualMode=*/tN2kGNSSDOPmode::N2kGNSSdm_3D,
+                          sbp_dops->hdop / 100.0,
+                          sbp_dops->vdop / 100.0,
+                          sbp_dops->tdop / 100.0);
+      } else {
+        SetN2kGNSSDOPData(N2kMsg, /*sequence ID=*/0xFF,
+                /*DesiredMode=*/tN2kGNSSDOPmode::N2kGNSSdm_Auto,
+                /*ActualMode=*/tN2kGNSSDOPmode::N2kGNSSdm_Error,
+                          N2kDoubleNA, N2kDoubleNA, N2kDoubleNA);
+      }
+      NMEA2000.SendMsg(N2kMsg);
+
+      d << "\tTOW: " << sbp_dops->tow << "\n"
+        << "\tHDOP: " << sbp_dops->hdop / 100.0 << "\n"
+        << "\tVDOP: " << sbp_dops->vdop / 100.0 << "\n"
+        << "\tTDOP: " << sbp_dops->tdop / 100.0 << "\n";
+
+      n2k_composite_message_cache.ResetIfOld(sbp_dops->tow);
+      n2k_composite_message_cache.SetFields(sbp_dops);
+      if (is_valid &&
+          n2k_composite_message_cache.IsReadyToSend(/*is_valid=*/true)) {
+        N2kMsg.Clear();
+        n2k_composite_message_cache.FillN2kMsg(&N2kMsg, /*is_valid=*/true);
+        NMEA2000.SendMsg(N2kMsg);
+      }
+
+      d << "\tDone.\n";
+    }
+
+    // PGN 129540
+    void callback_sbp_tracking_state(u16 sender_id, u8 len, u8 msg[],
+                                     void *context) {
+      UNUSED(sender_id);
+      UNUSED(context);
+      d << __FUNCTION__ << "\n"
+        << "\tGot " << static_cast<u16>(len) << "/"
+        << sizeof(tracking_channel_state_t) << "="
+        << static_cast<u16>(len / sizeof(tracking_channel_state_t))
+        << " states. Message len is most likely statically allocated.\n";
+
+      constexpr u8 cConstellationMaxSats = 32;
+      std::array<std::pair<u8, u8>, cConstellationMaxSats> sats_gps = {};
+      std::array<std::pair<u8, u8>, cConstellationMaxSats> sats_glo = {};
+      auto sats_gps_end_it = sats_gps.begin();
+      auto sats_glo_end_it = sats_glo.begin();
+      auto sbp_tracking_state = reinterpret_cast<msg_tracking_state_t*>(msg);
+      for(u8 i = 0; i < (len / sizeof(tracking_channel_state_t)); ++i) {
```

Comment:
Consider using `sizeof(*sbp_tracking_state->states)` so you don't have to explicitly reference the type name.